main/ZgatewayRF.ino

```diff
@@ -186,21 +191,63 @@ void MQTTtoRF(char* topicOri, JsonObject& RFdata) { // json object decoding
       Log.notice(F("RF Protocol:%d" CR), valuePRT);
       Log.notice(F("RF Pulse Lgth: %d" CR), valuePLSL);
       Log.notice(F("Bits nb: %d" CR), valueBITS);
+#    ifdef ZradioCC1101 // set Receive off and Transmitt on
+      float trMhz = RFdata["mhz"] | CC1101_FREQUENCY;
+      if (validFrequency(trMhz)) {
+        ELECHOUSE_cc1101.SetTx(trMhz);
+        Log.notice(F("RF mhz: %d" CR), (int)trMhz); // this is a float but no easy way to convert
+        mySwitch.disableReceive();
+        mySwitch.enableTransmit(RF_EMITTER_GPIO);
+      }
+#    endif
       mySwitch.setRepeatTransmit(valueRPT);
       mySwitch.setProtocol(valuePRT, valuePLSL);
       mySwitch.send(data, valueBITS);
       Log.notice(F("MQTTtoRF OK" CR));
       pub(subjectGTWRFtoMQTT, RFdata); // we acknowledge the sending by publishing the value to an acknowledgement topic, for the moment even if it is a signal repetition we acknowledge also
       mySwitch.setRepeatTransmit(RF_EMITTER_REPEAT); // Restore the default value
+#    ifdef ZradioCC1101 // set Receive off and Transmitt on
+      if (trMhz != valueMhz)
+        ELECHOUSE_cc1101.Init();
```

Comment:
Why is this call needed here?

Reply:
I found that when changing the receive frequency the module needed to be re-initialized in order for the change to take effect.  Checking the CC1101 data sheet and the examples, I couldn't find an example with this use case, so this is a bit brute force.